Updates to Google's updates on EWV and EFW deprecation (#74430)
Azure/azure-content: articles/active-directory/external-identities/google-federation.md

> Google federation is designed specifically for Gmail users. To federate with G Suite domains, use [direct federation](direct-federation.md).

> [!IMPORTANT]

> > **Starting in the second half of 2021**, Google is [deprecating WebView sign-in support](https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html). If you’re using Google federation for B2B invitations or B2C or self-service sign-up with Gmail, you should [test your line-of-business native applications for compatibility](google-federation.md#deprecation-of-webview-sign-in-support).

Reviewer: When using the term generically, looks like Google is using "web-view" (lowercase & hyphenated). I suggested that change throughout except where we were referring specifically to WebView controls - please have a look. 

```suggestion
> **Starting in the second half of 2021**, Google is [deprecating web-view sign-in support](https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html). If you’re using Google federation for B2B invitations or Azure AD B2C, or if you're using self-service sign-up with Gmail, you should [test your line-of-business native applications for compatibility](google-federation.md#deprecation-of-webview-sign-in-support).
```

> > **Starting later in 2021**, Google is [deprecating WebView sign-in support](https://developers.googleblog.com/2020/08/guidance-for-our-effort-to-block-less-secure-browser-and-apps.html). If you’re using Google federation for B2B invitations or B2C or self-service sign-up with Gmail, you should [test your line-of-business native applications for compatibility](google-federation.md#deprecation-of-webview-sign-in-support).

Reviewer 2: 2016 link.    
Also, second half of 2021?

## Deprecation of Embedded WebView and Framework sign-in support

Starting in the second half of 2021, Google is [deprecating embedded WebView sign-in support](https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html). If you’re using Google federation for B2B or [B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/identity-provider-google?pivots=b2c-user-flow) or [self-service sign-up with Gmail](identity-providers.md), you should test your line-of-business native applications for compatibility. If your apps authenticate users with an embedded webview, Google Gmail users won't be able to authenticate. 

The following are known scenarios that will impact Gmail users:
- Windows apps that use the [WebView](https://docs.microsoft.com/en-us/windows/communitytoolkit/controls/wpf-winforms/webview) control, [WebView2](https://docs.microsoft.com/en-us/microsoft-edge/webview2/), or the older WebBrowser control, for authentication. These apps should migrate to using the Web Account Manager (WAM) flow.
- Android applications using the WebView UI element 

> - iOS applications using the UIWebView/WKWebview 

Reviewer 2: Can remove "the"

You can also give Google guest users a direct link to an application or resource by including your tenant information, for example `https://myapps.microsoft.com/signin/Twitter/<application ID?tenantId=<your tenant ID>`.

> ## Deprecation of Embedded WebView and Framework sign-in support

Reviewer: Question: Can we shorten this to "Deprecation of embedded web-view sign-in support"? Since this level 2 heading also shows up in the mini table of contents on the page, it's good to keep the titles short (I adjusted bookmarks later on accordingly):

```suggestion
## Deprecation of embedded web-view support
```

> Starting in the second half of 2021, Google is [deprecating embedded WebView sign-in support](https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html). If you’re using Google federation for B2B or [B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/identity-provider-google?pivots=b2c-user-flow) or [self-service sign-up with Gmail](identity-providers.md), you should test your line-of-business native applications for compatibility. If your apps authenticate users with an embedded webview, Google Gmail users won't be able to authenticate. 

Reviewer: ```suggestion
Starting in the second half of 2021, Google is [deprecating embedded web-view sign-in support](https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html). If you’re using Google federation for B2B or [Azure AD B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/identity-provider-google?pivots=b2c-user-flow), or if you're using [self-service sign-up with Gmail](identity-providers.md), you should test your line-of-business native applications for compatibility. If your apps authenticate users with an embedded web-view, Google Gmail users won't be able to authenticate.

```

The following are known scenarios that will impact Gmail users:
- Windows apps that use the [WebView](https://docs.microsoft.com/en-us/windows/communitytoolkit/controls/wpf-winforms/webview) control, [WebView2](https://docs.microsoft.com/en-us/microsoft-edge/webview2/), or the older WebBrowser control, for authentication. These apps should migrate to using the Web Account Manager (WAM) flow.
- Android applications using the WebView UI element 
- iOS applications using the UIWebView/WKWebview 
- Apps using ADAL

This change does not affect:

- Windows apps
- Web apps

> - Mobile apps using system webviews for authentication ([SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) on iOS, [Custom Tabs](https://developer.chrome.com/docs/android/custom-tabs/overview/) on Android).  

Reviewer: ```suggestion
- Mobile apps using system web-views for authentication ([SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) on iOS, [Custom Tabs](https://developer.chrome.com/docs/android/custom-tabs/overview/) on Android).  
```

- G Suite identities, for example when you’re using SAML-based [direct federation](direct-federation.md) with G Suite

We’re confirming with Google whether this change affects the following:
- Windows apps that use the Web Account Manager (WAM) or Web Authentication Broker (WAB).  

We’re continuing to test various platforms and scenarios, and will update this article accordingly.

> ### Action needed for embedded webviews

Reviewer: ```suggestion

### Action needed for embedded web-views

```

> Modify your apps to use the system browser for sign-in. For details, see [Embedded vs System Web UI](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-web-browsers#embedded-vs-system-web-ui) in the MSAL.NET documentation. All MSAL SDKs use the system webview by default. 

Reviewer: ```suggestion
Modify your apps to use the system browser for sign-in. For details, see [Embedded vs System Web UI](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-web-browsers#embedded-vs-system-web-ui) in the MSAL.NET documentation. All MSAL SDKs use the system web-view by default.

```

### What to expect

> Before Google puts these changes into place in the second half of 2021, Microsoft will deploy a workaround for apps still using embedded webviews to ensure that Google does not block authentication. 

Reviewer: ```suggestion
Before Google puts these changes into place in the second half of 2021, Microsoft will deploy a workaround for apps still using embedded web-views to ensure that authentication isn't blocked.

```

> Applications that migrate to an allowed webview for authentication will not see this experience, and users will be allowed to authenticate via Google as normal. 

Reviewer: ```suggestion
Applications that are migrated to an allowed web-view for authentication won't be affected, and users will be allowed to authenticate via Google as usual. 
```

We will update this document as dates and further details are shared by Google. 

> ### Distinguishing between CEF/Electron and embedded webviews

Reviewer: ```suggestion
### Distinguishing between CEF/Electron and embedded web-views
```

> In addition to /Deprecation of WebView sign-in support/, Google is also [deprecating CEF-based Gmail authentication](https://developers.googleblog.com/2020/08/guidance-for-our-effort-to-block-less-secure-browser-and-apps.html). For applications built on CEF (Chromium Embedded Framework), such as Electron apps, Google will disable authentication disabled on June 30th 2021.  Impacted applications have received notice from Google directly, and are not covered in this documentation.  This document pertains to the embedded webviews described above, which Google will restrict at a separate date later in 2021. 

Reviewer: ```suggestion
In addition to the [deprecation of embedded web-view and framework sign-in support](#deprecation-of-embedded-web-view-sign-in-support), Google is also [deprecating CEF-based Gmail authentication](https://developers.googleblog.com/2020/08/guidance-for-our-effort-to-block-less-secure-browser-and-apps.html). For applications built on CEF (Chromium Embedded Framework), such as Electron apps, Google will disable authentication on June 30, 2021. Impacted applications have received notice from Google directly, and are not covered in this documentation.  This document pertains to the embedded web-views described above, which Google will restrict at a separate date later in 2021.
```
